app/models/ncr/work_order.rb

```diff
@@ -114,13 +114,23 @@ def approving_official
       self.approvers.first
     end
 
-    def current_approver_email_address
+    def current_approver
       if self.pending?
-        self.currently_awaiting_approvers.first.email_address
+        self.currently_awaiting_approvers.first
       elsif self.approving_official
-        self.approving_official.email_address
+        self.approving_official
+      elsif self.emergency and self.approvers.empty?
+        nil
       else
-        self.system_approver_emails.first
+        User.for_email(self.system_approver_emails.first)
+      end
+    end
+
+    def final_approver
+      if self.emergency and self.approvers.empty?
```

Comment:
how about:

```ruby
if not_emergency? && no_approvers?
  approvers.last
end
```

I think that's easier to read. Methods used in `if` would need to be defined in order for this to work..